proto/server/datastore/datastore.proto

```diff
@@ -2,348 +2,329 @@ syntax = "proto3";
+
+message AttestedNodeEntry {
+    // Node SPIFFE ID
+    string spiffe_id = 1;
+
+    // Attestation data type
+    string attestation_data_type = 2;
+
+    // Node certificate serial number
+    string cert_serial_number = 3;
+
+    // Node certificate not_after (seconds since unix epoch)
+    int64 cert_not_after = 4;
+}
+
+message CreateAttestedNodeEntryRequest {
+    AttestedNodeEntry entry = 1;
 }
 
+message CreateAttestedNodeEntryResponse {
+    AttestedNodeEntry entry = 1;
+}
 
-// Represents a Spiffe ID
-message FetchNodeResolverMapEntryRequest {
-    // SPIFFE ID
-    string baseSpiffeId = 1;
+message FetchAttestedNodeEntryRequest {
+    string spiffe_id = 1;
 }
 
-// Represents a list of Node resolver map entries for the specified Spiffe ID
-message FetchNodeResolverMapEntryResponse {
-    // List of Node resolver map entries
-    repeated NodeResolverMapEntry nodeResolverMapEntryList = 1;
+message FetchAttestedNodeEntryResponse {
+    AttestedNodeEntry entry = 1;
 }
 
+message ListAttestedNodeEntriesRequest {
+    google.protobuf.Int64Value by_expires_before = 1;
+}
 
-// Represents a Node resolver map entry to delete
-message DeleteNodeResolverMapEntryRequest {
-    // Node resolver map entry
-    NodeResolverMapEntry nodeResolverMapEntry = 1;
+message ListAttestedNodeEntriesResponse {
+    repeated AttestedNodeEntry entries = 1;
 }
 
-// Represents a list of Node resolver map entries
-message DeleteNodeResolverMapEntryResponse {
-    // List of Node resolver map entries
-    repeated NodeResolverMapEntry nodeResolverMapEntryList = 1;
+message UpdateAttestedNodeEntryRequest {
+    string spiffe_id = 1;
+
+    string cert_serial_number = 2;
+
+    int64 cert_not_after = 3;
 }
 
+message UpdateAttestedNodeEntryResponse {
+    AttestedNodeEntry entry = 1;
+}
 
-// Represents a list of Node resolver map entries
-message RectifyNodeResolverMapEntriesRequest {
-    // List of Node resolver map entries
-    repeated NodeResolverMapEntry nodeResolverMapEntryList = 1;
+message DeleteAttestedNodeEntryRequest {
+    string spiffe_id = 1;
 }
 
-// Represents a list of Node resolver map entries
-message RectifyNodeResolverMapEntriesResponse {
-    // List of Node resolver map entries
-    repeated NodeResolverMapEntry nodeResolverMapEntryList = 1;
+message DeleteAttestedNodeEntryResponse {
+    AttestedNodeEntry entry = 1;
 }
 
-//
-//
-//
 
-// Represents a Registration entry to create
+/////////////////////////////////////////////////////////////////////////////
+// Registration Entries
+/////////////////////////////////////////////////////////////////////////////
+
 message CreateRegistrationEntryRequest {
-    // Registration entry
-    spire.common.RegistrationEntry registeredEntry = 1;
+    spire.common.RegistrationEntry entry = 1;
 }
 
-// Represents the created Registration entry
 message CreateRegistrationEntryResponse {
-    // Registration entry ID
-    string registeredEntryId = 1;
+    string entry_id = 1;
```

Comment:
Since the time this message was created, `entry_id` has been added to the `spire.common.RegistrationEntry` message... perhaps we can return it here instead?

Reply:
I'd be super happy with that. That cleans up all sorts of code.